### PR TITLE
Op 18676

### PR DESCRIPTION
**Pull request by Pranav-b-7**
We prefer small, well tested pull requests.

Please refer to [Contributing to Spinnaker](https://spinnaker.io/community/contributing/).

When filling out a pull request, please consider the following:

* Follow the commit message conventions [found here](https://spinnaker.io/community/contributing/submitting/).
* Provide a descriptive summary for your changes.
* If it fixes a bug or resolves a feature request, be sure to link to that issue.
* Add inline code comments to changes that might not be obvious.
* Squash your commits as you keep adding changes.
* Add a comment to @spinnaker/reviewers for review if your issue has been outstanding for more than 3 days.

Note that we are unlikely to accept pull requests that add features without prior discussion. The best way to propose a feature is to open an issue first and discuss your ideas there before implementing them.

https://devopsmx.atlassian.net/browse/OP-18676

### CODE REVIEW

File: docker_build/gate.yml [hunk 1]
```diff
@@ -121,3 +121,217 @@ management:
     health:
       show-details: always
       show-components: always
+
+feature:
```
Comment (rahul-chekuri) [inline docker_build/gate.yml:125]: let us have dedicated config files for this. Moving forward, we will add many fields that would increase this file's size.

File: gate-web/src/main/groovy/com/netflix/spinnaker/gate/config/GateConfig.groovy [hunk 2]
```diff
@@ -522,15 +521,4 @@ class GateConfig extends RedisHttpSessionConfiguration {
                                                   FiatClientConfigurationProperties fiatClientConfigurationProperties) {
     return new FiatPermissionEvaluator(registry, fiatService, fiatClientConfigurationProperties, fiatStatus)
   }
-    @Component
```
Comment (rahul-chekuri) [inline gate-web/src/main/groovy/com/netflix/spinnaker/gate/config/GateConfig.groovy:525]: why is this removed ?

File: gate-web/src/main/java/com/opsmx/spinnaker/gate/model/KeycloakAuthProvider.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+@Data
+@Configuration
+@ConfigurationProperties(prefix = "keycloak.auth.providers")
+@ConditionalOnExpression("${feature.auth-provider.flag:false}")
+public class KeycloakAuthProvider {
+
+  private AuthParam ldap;
+  private AuthParam saml;
+
```
Comment (rahul-chekuri) [inline gate-web/src/main/java/com/opsmx/spinnaker/gate/model/KeycloakAuthProvider.java:33]: Should this be a list? tomorrow if we want to include a new auth type we should change the code again? we wanted this to be purely configurable right?

**Review by rahul-chekuri — COMMENTED** [review]
1. Add logs/comments wherever it is required
2. is an RBAC check required?
3. add unit tests; should cover the required fields at least; create a sub-task if you want to take this in the end.
4. Add a feature flag

File: gate-web/src/main/groovy/com/netflix/spinnaker/gate/config/GateConfig.groovy (hunk 2 above)
Comment (Pranav-b-7) [reply gate-web/src/main/groovy/com/netflix/spinnaker/gate/config/GateConfig.groovy:525]: Hystrix code and dependency is removed in OSS, so i have done the same thing here as well otherwise the build will fail

File: gate-web/src/main/java/com/opsmx/spinnaker/gate/model/KeycloakAuthProvider.java (hunk 3 above)
Comment (Pranav-b-7) [reply gate-web/src/main/java/com/opsmx/spinnaker/gate/model/KeycloakAuthProvider.java:33]: Made changes as suggested

File: docker_build/gate.yml (hunk 1 above)
Comment (Pranav-b-7) [reply docker_build/gate.yml:125]: Made changes as suggested

**Comment by Pranav-b-7** [comment]
> 1. Add logs/comments wherever it is required
> 2. is an RBAC check required?
> 3. add unit tests; should cover the required fields at least; create a sub-task if you want to take this in the end.
> 4. Add a feature flag

Unit test case and RBAC check shall be added later on

**Review by rahul-chekuri — APPROVED** [review]
